src/github.com/matrix-org/dendrite/roomserver/storage/sql.go

```diff
@@ -326,20 +363,46 @@ func (s *statements) prepareEvents(db *sql.DB) (err error) {
 	if s.insertEventStmt, err = db.Prepare(insertEventSQL); err != nil {
 		return
 	}
+	if s.selectStateEventsByIDStmt, err = db.Prepare(selectStateEventsByIDSQL); err != nil {
+		return
+	}
 	return
 }
 
 func (s *statements) insertEvent(
 	roomNID, eventTypeNID, eventStateKeyNID int64,
 	eventID string,
 	referenceSHA256 []byte,
+	authEventNIDs []int64,
 ) (eventNID int64, err error) {
 	err = s.insertEventStmt.QueryRow(
 		roomNID, eventTypeNID, eventStateKeyNID, eventID, referenceSHA256,
+		pq.Int64Array(authEventNIDs),
 	).Scan(&eventNID)
 	return
 }
 
+func (s *statements) selectStateEventsByID(eventIDs []string) ([]types.StateEntry, error) {
+	results := make([]types.StateEntry, len(eventIDs))
+	rows, err := s.selectStateEventsByIDStmt.Query(pq.StringArray(eventIDs))
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	i := 0
+	for ; rows.Next(); i++ {
```

Comment:
Can you do this [here](https://github.com/matrix-org/dendrite/pull/4/files#diff-8e2932ebbde743c3da4d0d9a88d5b205R268) please? Be consistent, and I prefer the in-line form.